web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { Flex, Indicator, Box } from 'design';
+
+import styled from 'styled-components';
+
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import ErrorMessage from 'teleport/components/AgentErrorMessage';
+import useTeleport from 'teleport/useTeleport';
+
+import { useResources } from './useResources';
+import { ResourceCard } from './ResourceCard';
+
+export function Resources() {
+  const teleCtx = useTeleport();
+  const { attempt, fetchedData, fetchMore } = useResources(teleCtx);
+  const observed = React.useRef(null);
+
+  React.useEffect(() => {
+    if (observed.current) {
+      const observer = new IntersectionObserver(entries => {
+        if (entries[0]?.isIntersecting) {
+          fetchMore();
+        }
+      });
+      observer.observe(observed.current);
+      return () => observer.disconnect();
+    }
+  });
```

Comment:
this is missing a dependency array, which should only require `fetchMore`. Also, this will cause `fetchMore` to run even on the initial load, which we don't want. (These are the times I miss the old class component lifecycle methods). I've seen a pattern before with useEffect where you want it to run on every update to the dependency array but NOT on initial load by using a `isMounted = useRef(false)` type of thing, then setting it to true once you've received the first bit of data or something.

I don't know if I've run into this as issue but just off the top of my head, should we also be checking for `attempt.status !== "processing"` before we 'fetchMore' in this conditional?

Reply:
And that's a very good catch. This condition may actually fix a bug I saw earlier today.

Reply:
So about checking for attempt status: the `fetchMore` function already does it internally. I added the missing dependencies, but didn't fix the bug I was talking about ;) I'll fix this one separately.